third_party/ibis/ibis_teradata/registry.py

```diff
@@ -226,6 +226,24 @@ def _string_join(translator, op):
     return "||".join(map(translator.translate, op.arg))
 
 
+def _extract_epoch(translator, op):
+    arg = translator.translate(op.arg)
+    if op.arg.output_dtype.timezone:
```

Comment:
Minor thought but this might be more concise as:
```
utc_expression = "AT TIME ZONE 'GMT'" if op.arg.output_dtype.timezone else ""
return (
    f"(CAST({arg} AS DATE {utc_expression}) - DATE '1970-01-01') * 86400 + "
    f"(EXTRACT(HOUR FROM {arg} {utc_expression}) * 3600) + "
    f"(EXTRACT(MINUTE FROM {arg} {utc_expression}) * 60) + "
    f"(EXTRACT(SECOND FROM {arg} {utc_expression}))"
)
```
Minor enough to ignore though.